src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleasePermissionService.cs

```diff
@@ -0,0 +1,181 @@
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using static GovUk.Education.ExploreEducationStatistics.Content.Model.ReleaseRole;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services
+{
+    public class ReleasePermissionService : IReleasePermissionService
+    {
+        private readonly ContentDbContext _contentDbContext;
+        private readonly IPersistenceHelper<ContentDbContext> _persistenceHelper;
+        private readonly IUserReleaseRoleRepository _userReleaseRoleRepository;
+        private readonly IUserService _userService;
+
+        public ReleasePermissionService(
+            ContentDbContext contentDbContext,
+            IPersistenceHelper<ContentDbContext> persistenceHelper,
+            IUserReleaseRoleRepository userReleaseRoleRepository,
+            IUserService userService)
+        {
+            _contentDbContext = contentDbContext;
+            _persistenceHelper = persistenceHelper;
+            _userReleaseRoleRepository = userReleaseRoleRepository;
+            _userService = userService;
+        }
+
+        public async Task<Either<ActionResult, List<ContributorViewModel>>>
+            GetReleaseContributorPermissions(Guid publicationId, Guid releaseId)
+        {
+            return await _persistenceHelper
+                .CheckEntityExists<Publication>(publicationId,
+                    query =>
+                        query.Include(r => r.Releases))
+                .OnSuccessDo(publication => _userService
+                    .CheckCanUpdateReleaseRole(publication, Contributor))
+                .OnSuccess(async publication =>
+                {
+                    var release = publication.Releases.SingleOrDefault(r => r.Id == releaseId);
+
+                    var allContributorsForRelease = await _contentDbContext.UserReleaseRoles
+                        .Include(urr => urr.User)
+                        .ToAsyncEnumerable()
+                        .Where(urr =>
+                            urr.ReleaseId == release?.Id
+                            && urr.Role == Contributor)
+                        .ToListAsync();
+
+                    return allContributorsForRelease
+                        .Select(urr => new ContributorViewModel
+                        {
+                            ReleaseRoleId = urr.Id,
+                            UserId = urr.UserId,
+                            UserFullName = urr.User.DisplayName,
+                            UserEmail = urr.User.Email,
+                        })
+                        .OrderBy(model => model.UserFullName)
+                        .ToList();
+                });
+        }
+
+        public async Task<Either<ActionResult, List<ContributorViewModel>>>
+            GetPublicationContributorList(Guid releaseId)
```

Comment:
It doesn't really make sense for this to receive a `releaseId`. We should be working with the `publicationId`. This also relates to my comment lower down about `releaseRoleId` not making sense on the returned list of contributors.